File: python/y402/core/types/schema.py
from typing import Optional, Dict, Literal, Any
from pydantic import BaseModel, ConfigDict
from pydantic.alias_generators import to_camel


class HTTPInputSchema(BaseModel):
    """
    Schema for HTTP request input, excluding spec
    and method which are handled by the middleware.
    """

    query_params: Optional[Dict[str, str]] = None
    body_type: Optional[
        Literal["json", "form-data", "multipart-form-data", "text", "binary"]
    ] = None
    body_fields: Optional[Dict[str, Any]] = None
    header_fields: Optional[Dict[str, Any]] = None

    model_config = ConfigDict(
        alias_generator=to_camel,
        populate_by_name=True,
        from_attributes=True,
    )

File: ts/y402/src/core/types/schema.ts
import { z } from "zod";

/**
 * Schema for HTTP request input, excluding spec
 * and method which are handled by the middleware.
 */
export const HTTPInputSchema = z.object({
    queryParams: z.record(z.string()).optional(),
    bodyType: z
        .union([
            z.literal("json"),
            z.literal("form-data"),
            z.literal("multipart-form-data"),
            z.literal("text"),
            z.literal("binary"),
        ])
        .optional(),
    bodyFields: z.record(z.unknown()).optional(),
    headerFields: z.record(z.unknown()).optional(),
});

export type HTTPInput = z.infer<typeof HTTPInputSchema>;
